src/hooks/useKeyPair.ts

```diff
@@ -132,27 +135,59 @@ const setKeyPair = async ({
      * Ignoring this error is fine, since we are falling back to just storing it in
      * memory.
      */
-    await setKeyPairToIdb(userId, storedKeyPair).catch(() => {})
+    await setKeyPairToIdb(newUser.id, storedKeyPair).catch(() => {})
   }
 
-  await mutate(`/user/${account}`)
+  await mutate(
```

Comment:
Do we need these awaits here, or the mutation will happen immediately since we're passing the new data in the second param? If we need to await each mutation, I wonder if we could use Promise.all, to do it all at once?

Reply:
Wow, nice! I never thought about this, but just checked, and it returns an instantly _fulfilled_ state Promise, so I think you are right, and the mutation itself is synchronous! Since we don't need the return values here, I've removed the awaits in https://github.com/agoraxyz/guild.xyz/pull/838/commits/a5e6f03522c34ee5e786535c044074ebf07c3be0!
I've tested it out, and seems to work as expected!
This is so cool, thanks for pointing it out!